version: "3.9"
services:
  jupyter:
    depends_on:
      - pg
    image: tensorflow_jupyter
    ports:
      - "62000:8888"
    volumes:
      - ./:/home/jovyan/app:Z
    runtime: nvidia
    devices:
      - /dev/nvidia0:/dev/nvidia0
      - /dev/nvidiactl:/dev/nvidiactl
      - /dev/nvidia-modeset:/dev/nvidia-modeset
      - /dev/nvidia-uvm:/dev/nvidia-uvm
      - /dev/nvidia-uvm-tools:/dev/nvidia-uvm-tools
    command: jupyter notebook --ip=0.0.0.0 --no-browser
    # restart: on-failure:10

   
  pg:
    image: postgres:latest
    ports: 
      - "5432:5432"
    volumes:
      - pg_temperatura:/var/lib/postgresql/data:Z
    environment:
      - POSTGRES_HOST_AUTH_METHOD=trust

  pg_brasil:
    image: postgis/postgis:latest
    volumes:
      - pg_brasil:/var/lib/postgresql/data:Z
    environment:
      - POSTGRES_HOST_AUTH_METHOD=trust

volumes:
  pg_temperatura:
    external: true
    name: pg_temperatura
  pg_brasil:
    external: true
    name: pg_brasil
